src/main/java/com/azure/spring/dev/tools/dependency/support/SpringProjectMetadataReader.java

```diff
@@ -43,9 +46,12 @@ public String getCurrentVersion() {
             .stream()
             .filter(p -> p.getReleaseStatus().equals(ReleaseStatus.GENERAL_AVAILABILITY))
             .filter(p -> p.getVersion().matches("2\\.\\d\\.\\d+"))
-            .filter(Objects::nonNull)
             .map(ProjectRelease::getVersion)
+            .map(DEFAULT::parse)
```

Comment:
After a second thought, I think `VersionParser.DEFAULT::parse` is better, and we could remove the static import.